--- git/MSL/Complex.mo	2016-02-26 17:04:36.331995480 +0100
+++ build/Complex trunk.mo	2016-02-26 17:12:55.789524627 +0100
@@ -1,5 +1,7 @@
 within ;
+
 operator record Complex "Complex number with overloaded operators"
+
 //record Complex "Complex number with overloaded operators"
   replaceable Real re "Real part of complex number" annotation(Dialog);
   replaceable Real im "Imaginary part of complex number" annotation(Dialog);
@@ -249,7 +251,7 @@
 
 
 annotation (Protection(access=Access.hide),
-version="3.2.2",
+version= "trunk",
 versionBuild=0,
 versionDate="2016-01-15",
 dateModified = "2016-01-15 08:44:41Z",
@@ -270,5 +272,4 @@
           lineColor={255,255,255},
           extent={{-90,-50},{90,50}},
           textString="C")}));
-
-end Complex;
+end Complex;
\ No newline at end of file
