diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2018-06-11 19:09:06.694009939 +0200
+++ build/ModelicaTest trunk/package.mo	2018-06-12 19:02:21.679099497 +0200
@@ -49,12 +49,12 @@
 end testAllFunctions;
 
   annotation (preferredView="info",
-       version="3.2.3",
+       version= "trunk",
        versionBuild=1,
        versionDate="2018-08-01",
        dateModified = "2018-04-20 12:00:00Z",
        revisionId="$Format:%h %ci$",
-       uses(Modelica(version="3.2.3")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
