diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaServices/package.mo "build/ModelicaServices trunk/package.mo"
--- git/Modelica/ModelicaServices/package.mo	2018-12-04 19:02:04.356660245 +0100
+++ build/ModelicaServices trunk/package.mo	2018-12-04 19:02:26.064968887 +0100
@@ -1,7 +1,7 @@
 within ;
-package ModelicaServices "ModelicaServices (Default implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
+package ModelicaServices "ModelicaServices (OpenModelica implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
   extends Modelica.Icons.Package;
-  constant String target="Default"
+  constant String target= "OpenModelica"
     "Target of this ModelicaServices implementation";
 
   package UsersGuide "User's Guide"
@@ -144,7 +144,7 @@
       extends
         Modelica.Utilities.Internal.PartialModelicaServices.ExternalReferences.PartialLoadResource;
       algorithm
-      fileReference := Modelica.Utilities.Files.fullPathName(uri);
+      fileReference:=OpenModelica.Scripting.uriToFilename(uri);
 
       annotation (Documentation(info="<html>
 <p>
@@ -162,7 +162,7 @@
       "Smallest number such that small and -small are representable on the machine";
     final constant Real inf=1e60
       "Biggest Real number such that inf and -inf are representable on the machine";
-    final constant Integer Integer_inf=2147483647
+    final constant Integer Integer_inf=OpenModelica.Internal.Architecture.integerMax()
       "Biggest Integer number such that Integer_inf and -Integer_inf are representable on the machine";
     annotation (Documentation(info="<html>
 <p>
@@ -211,12 +211,12 @@
 
   annotation (
     preferredView="info",
-    version="3.2.3",
+    version= "trunk",
     versionBuild=1,
     versionDate="2018-09-26",
     dateModified = "2018-09-26 12:00:00Z",
     revisionId="$Format:%h %ci$",
-    uses(Modelica(version="3.2.3")),
+    uses(Modelica(version= "trunk")),
     conversion(
       noneFromVersion="1.0",
       noneFromVersion="1.1",
