diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2018-01-15 19:03:49.761135045 +0100
+++ build/ModelicaTest trunk/package.mo	2018-01-15 19:04:52.337558647 +0100
@@ -51,12 +51,12 @@
 
 
   annotation (preferredView="info",
-       version="3.2.2",
+       version= "trunk",
        versionBuild=0,
        versionDate="2016-03-11",
        dateModified = "2016-03-11 08:44:41Z",
        revisionId="$Id::                                       $",
-       uses(Modelica(version="3.2.2")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
