diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2020-03-01 19:14:03.970501582 +0100
+++ build/ModelicaTest trunk/package.mo	2020-03-02 19:04:25.728464957 +0100
@@ -51,11 +51,11 @@
 end testAllFunctions;
 
   annotation (preferredView="info",
-       version="4.0.0",
+       version= "trunk",
        versionDate="2020-02-12",
        dateModified = "2020-02-12 19:00:00Z",
        revisionId="$Format:%h %ci$",
-       uses(Modelica(version="4.0.0")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
