diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2020-05-11 19:03:29.179428753 +0200
+++ build/ModelicaTest trunk/package.mo	2020-05-11 19:03:49.247606623 +0200
@@ -51,11 +51,11 @@
 end testAllFunctions;
 
   annotation (preferredView="info",
-       version="4.0.0",
+       version= "trunk",
        versionDate="2020-05-04",
        dateModified = "2020-05-04 19:00:00Z",
        revisionId="$Format:%h %ci$",
-       uses(Modelica(version="4.0.0")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
