diff -u -x .svn -x .git -x Library -r git/Modelica/Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Transformer.mo "build/Modelica trunk/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Transformer.mo"
--- git/Modelica/Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Transformer.mo	2019-08-28 19:16:29.719798182 +0200
+++ build/Modelica trunk/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Transformer.mo	2019-08-29 19:03:38.313372969 +0200
@@ -270,5 +270,6 @@
                   fillColor={255,255,170},
                   fillPattern=FillPattern.Solid,
                   textStyle={TextStyle.Bold},
-          textString="%m phase transient")}));
+          textString="%m phase transient")}),
+  __OpenModelica_simulationFlags(nls = "newton"));
 end IMC_Transformer;
diff -u -x .svn -x .git -x Library -r git/Modelica/Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_YD.mo "build/Modelica trunk/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_YD.mo"
--- git/Modelica/Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_YD.mo	2019-08-28 19:16:29.719798182 +0200
+++ build/Modelica trunk/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_YD.mo	2019-08-29 19:03:38.133374093 +0200
@@ -199,5 +199,6 @@
                   fillColor={255,255,170},
                   fillPattern=FillPattern.Solid,
                   textStyle={TextStyle.Bold},
-                  textString="%m phase transient")}));
+                  textString="%m phase transient")}),
+  __OpenModelica_simulationFlags(nls = "newton"));
 end IMC_YD;
diff -u -x .svn -x .git -x Library -r git/Modelica/Modelica/package.mo "build/Modelica trunk/package.mo"
--- git/Modelica/Modelica/package.mo	2019-08-28 19:16:29.771798108 +0200
+++ build/Modelica trunk/package.mo	2019-08-29 19:03:39.141367808 +0200
@@ -8591,12 +8591,12 @@
 end UsersGuide;
 annotation (
 preferredView="info",
-version="4.0.0-dev",
+version= "trunk",
 versionBuild=1,
 versionDate="2019-01-23",
 dateModified = "2019-01-23 12:00:00Z",
 revisionId="$Format:%h %ci$",
-uses(Complex(version="4.0.0-dev"), ModelicaServices(version="4.0.0-dev")),
+uses(Complex(version= "trunk"), ModelicaServices(version= "trunk")),
 conversion(
  from(version={"3.0", "3.0.1", "3.1", "3.2", "3.2.1", "3.2.2", "3.2.3"}, script="modelica://Modelica/Resources/Scripts/Conversion/ConvertModelica_from_3.2.3_to_4.0.0.mos")),
 Icon(coordinateSystem(extent={{-100.0,-100.0},{100.0,100.0}}), graphics={
