diff -u -x .svn -r libraries/Modelica 3.2.1/ModelicaServices/package.mo build/ModelicaServices 3.2.1/package.mo
--- libraries/Modelica 3.2.1/ModelicaServices/package.mo	2013-02-18 14:23:52.168727647 +0100
+++ build/ModelicaServices 3.2.1/package.mo	2013-02-21 15:52:11.689485500 +0100
@@ -1,7 +1,7 @@
 within ;
-package ModelicaServices "ModelicaServices (Default implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
+package ModelicaServices "ModelicaServices (OpenModelica implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
 extends Modelica.Icons.Package;
-constant String target="Default"
+constant String target="OpenModelica"
   "Target of this ModelicaServices implementation";
 
 
@@ -692,7 +692,7 @@
     extends
       Modelica.Utilities.Internal.PartialModelicaServices.ExternalReferences.PartialLoadResource;
     algorithm
-    fileReference := Modelica.Utilities.Files.fullPathName(uri);
+      fileReference:=OpenModelica.Scripting.uriToFilename(uri);
 
     annotation (Documentation(info="<html>
 <p>
@@ -712,7 +712,7 @@
     "Smallest number such that small and -small are representable on the machine";
   final constant Real inf=1.e+60
     "Biggest Real number such that inf and -inf are representable on the machine";
-  final constant Integer Integer_inf=2147483647
+  final constant Integer Integer_inf=OpenModelica.Internal.Architecture.integerMax()
     "Biggest Integer number such that Integer_inf and -Integer_inf are representable on the machine";
   annotation (Documentation(info="<html>
 <p>
