diff -u -x .svn -x .git -x Library -r git/MSL/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/MSL/ModelicaTest/package.mo	2016-03-14 17:01:05.593070380 +0100
+++ build/ModelicaTest trunk/package.mo	2016-03-14 17:14:40.857696941 +0100
@@ -1,4 +1,5 @@
 within ;
+
 package ModelicaTest "Library to test components of package Modelica - Version 3.2.2"
 extends Modelica.Icons.Package;
 
@@ -51,12 +52,12 @@
 
 
   annotation (preferredView="info",
-       version="3.2.2",
+       version= "trunk",
        versionBuild=0,
        versionDate="2016-03-11",
        dateModified = "2016-03-11 08:44:41Z",
        revisionId="$Id::                                       $",
-       uses(Modelica(version="3.2.2")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
@@ -96,4 +97,4 @@
           lineColor={75,138,73},
           fillColor={75,138,73},
           fillPattern=FillPattern.Solid)}));
-end ModelicaTest;
+end ModelicaTest;
\ No newline at end of file
