diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2020-03-27 19:13:13.891698315 +0100
+++ build/ModelicaTest trunk/package.mo	2020-03-28 19:04:02.499839309 +0100
@@ -51,11 +51,11 @@
 end testAllFunctions;
 
   annotation (preferredView="info",
-       version="4.0.0",
+       version= "trunk",
        versionDate="2020-03-11",
        dateModified = "2020-03-11 19:00:00Z",
        revisionId="$Format:%h %ci$",
-       uses(Modelica(version="4.0.0")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
