diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaTest/package.mo "build/ModelicaTest trunk/package.mo"
--- git/Modelica/ModelicaTest/package.mo	2019-03-06 19:03:36.306313693 +0100
+++ build/ModelicaTest trunk/package.mo	2019-03-06 19:03:56.510442783 +0100
@@ -49,12 +49,12 @@
 end testAllFunctions;
 
   annotation (preferredView="info",
-       version="4.0.0-dev",
+       version= "trunk",
        versionBuild=1,
        versionDate="2019-01-23",
        dateModified = "2019-01-23 12:00:00Z",
        revisionId="$Format:%h %ci$",
-       uses(Modelica(version="4.0.0-dev")),
+       uses(Modelica(version= "trunk")),
     Documentation(info="<html>
 <p>
 This library provides models and functions to test components of
