diff -u -x .svn -x .git -x Library -r git/Modelica/ModelicaServices/package.mo "build/ModelicaServices trunk/package.mo"
--- git/Modelica/ModelicaServices/package.mo	2017-10-28 19:05:03.051398992 +0200
+++ build/ModelicaServices trunk/package.mo	2017-10-28 19:05:19.759526095 +0200
@@ -1,7 +1,7 @@
 within ;
-package ModelicaServices "ModelicaServices (Default implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
+package ModelicaServices "ModelicaServices (OpenModelica implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
   extends Modelica.Icons.Package;
-  constant String target="Default"
+  constant String target= "OpenModelica"
     "Target of this ModelicaServices implementation";
 
   package UsersGuide "User's Guide"
@@ -595,7 +595,7 @@
       extends
         Modelica.Utilities.Internal.PartialModelicaServices.ExternalReferences.PartialLoadResource;
       algorithm
-      fileReference := Modelica.Utilities.Files.fullPathName(uri);
+      fileReference:=OpenModelica.Scripting.uriToFilename(uri);
 
       annotation (Documentation(info="<html>
 <p>
@@ -613,7 +613,7 @@
       "Smallest number such that small and -small are representable on the machine";
     final constant Real inf=1e60
       "Biggest Real number such that inf and -inf are representable on the machine";
-    final constant Integer Integer_inf=2147483647
+    final constant Integer Integer_inf=OpenModelica.Internal.Architecture.integerMax()
       "Biggest Integer number such that Integer_inf and -Integer_inf are representable on the machine";
     annotation (Documentation(info="<html>
 <p>
@@ -663,12 +663,12 @@
   annotation (
     Protection(access=Access.hide),
     preferredView="info",
-    version="3.2.2",
+    version= "trunk",
     versionBuild=0,
     versionDate="2016-01-15",
     dateModified = "2016-01-15 08:44:41Z",
     revisionId="$Id::                                       $",
-    uses(Modelica(version="3.x.x")),
+    uses(Modelica(version= "trunk")),
     conversion(
       noneFromVersion="1.0",
       noneFromVersion="1.1",
