diff -u -x .svn -x .git -x Library -r git/MSL/ModelicaServices/package.mo "build/ModelicaServices 3.2.2/package.mo"
--- git/MSL/ModelicaServices/package.mo	2016-04-18 09:20:39.458162302 +0200
+++ build/ModelicaServices 3.2.2/package.mo	2016-04-18 09:21:13.561159970 +0200
@@ -1,7 +1,8 @@
 within ;
-package ModelicaServices "ModelicaServices (Default implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
+
+package ModelicaServices "ModelicaServices (OpenModelica implementation) - Models and functions used in the Modelica Standard Library requiring a tool specific implementation"
 extends Modelica.Icons.Package;
-constant String target="Default"
+constant String target= "OpenModelica"
   "Target of this ModelicaServices implementation";
 
 
@@ -622,7 +623,7 @@
     extends
       Modelica.Utilities.Internal.PartialModelicaServices.ExternalReferences.PartialLoadResource;
     algorithm
-    fileReference := Modelica.Utilities.Files.fullPathName(uri);
+      fileReference:=OpenModelica.Scripting.uriToFilename(uri);
 
     annotation (Documentation(info="<html>
 <p>
@@ -642,7 +643,7 @@
     "Smallest number such that small and -small are representable on the machine";
   final constant Real inf=1.e+60
     "Biggest Real number such that inf and -inf are representable on the machine";
-  final constant Integer Integer_inf=2147483647
+  final constant Integer Integer_inf=OpenModelica.Internal.Architecture.integerMax()
     "Biggest Integer number such that Integer_inf and -Integer_inf are representable on the machine";
   annotation (Documentation(info="<html>
 <p>
@@ -743,4 +744,4 @@
 </p>
 
 </html>"));
-end ModelicaServices;
+end ModelicaServices;
\ No newline at end of file
